listeners.py

```diff
@@ -186,14 +190,30 @@ def process_result(self, obj):
     def process_timeout(self, delta_seconds):
         self.warned_seconds_ago += delta_seconds
         self.inactivity += delta_seconds
+        self.set_task_for_worker_id()
         worker_ids = self.get_not_done_worker_ids()
         if self.warned_seconds_ago < self.warn_timeout:
             return
         color_stdout("No output during %d seconds. "
-            "List of workers not reporting the status: %s; "
-            "Will abort after %d seconds without output.\n" % (
-                self.inactivity, worker_ids, self.kill_timeout),
+            "Will abort after %d seconds without output. "
+            "List of workers not reporting the status:\n" % (
+                self.inactivity, self.kill_timeout),
                 schema='test_var')
+        for worker_id in worker_ids:
+            task_name, task_param = self.get_task_by_worker_id(worker_id)
+            color_stdout("[{0:03d}] {1} {2}\n".format(worker_id,
+                                                  task_name,
+                                                  task_param or ''),
+                         schema='test_var')
+            task_path = "{0:03d}_{1}".format(worker_id,
```

Comment:
I think it is better to get a path to result file from a worker and don’t try to reconstruct it here.